Sources/TuistDependencies/Carthage/Generators/CarthageCommandGenerator.swift

```diff
@@ -9,16 +9,17 @@ public protocol CarthageCommandGenerating {
     /// Builds `Carthage` command.
     /// - Parameters:
     ///   - path: Directory whose project's dependencies will be installed.
+    ///   - produceXCFrameworks: Indicates whether the `Carthage` produces XCFrameworks instead of universal frameworks.
     ///   - platforms: The platforms to build for.
-    func command(path: AbsolutePath, platforms: Set<Platform>?) -> [String]
+    func command(path: AbsolutePath, produceXCFrameworks: Bool, platforms: Set<Platform>?) -> [String]
 }
 
 // MARK: - Carthage Command Generator
 
 public final class CarthageCommandGenerator: CarthageCommandGenerating {
     public init() {}
 
-    public func command(path: AbsolutePath, platforms: Set<Platform>?) -> [String] {
+    public func command(path: AbsolutePath, produceXCFrameworks: Bool, platforms: Set<Platform>?) -> [String] {
```

Comment:
nitpick: maybe something `CarthageFrameworkOutputType` might be a little bit more descriptive and will be more extensible than a simple boolean. But it is fine as is, too